### PR TITLE
fix: Call logs obfuscation

### PR DESCRIPTION
**Pull request by pwnsdx**

### CODE REVIEW

File: app/script/telemetry/calling/CallLogger.js [hunk 1]
```diff
@@ -187,7 +180,9 @@ z.telemetry.calling.CallLogger = class CallLogger {
     const shouldLogToMemory = logLevel !== CallLogger.LOG_LEVEL.OFF;
     if (shouldLogToMemory) {
       const logType = this.getDebugType(logLevel);
-      const logMessage = `[${new Date().toISOString()}] [${this.name}] (${logType}) ${obfuscatedMessage}`;
+      const logMessage = this.safeGuard(
+        `[${new Date().toISOString()}] [${this.name}] (${logType}) ${obfuscatedMessage}`
+      );
```
Comment (gregor) [inline app/script/telemetry/calling/CallLogger.js]: Break this up if the function call does not fit into a single line.

File: app/script/telemetry/calling/CallLogger.js [hunk 2]
```diff
@@ -244,4 +239,21 @@ z.telemetry.calling.CallLogger = class CallLogger {
...
+      return this.obfuscate(match);
+    });
+
+    // Ensure IP addresses are obfuscated
+    message = message.replace(CallLogger.REGEXES.IPV4, match => {
+      return this.obfuscateIp(match);
+    });
+    message = message.replace(CallLogger.REGEXES.IPV6, match => {
+      return this.obfuscateIp(match);
+    });
```
Comment (gregor) [inline app/script/telemetry/calling/CallLogger.js]: My hunch is that these would fit into a single line.

File: test/unit_tests/telemetry/calling/CallLoggerSpec.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+// grunt test_init && grunt test_run:telemetry/calling/CallLogger
+
+describe('z.telemetry.calling.CallLogger', () => {
+  it('properly replaces IP addresses', () => {
```
Comment (bennycode) [inline test/unit_tests/telemetry/calling/CallLoggerSpec.js]: BIG 👍 for adding tests. 🍾

**Comment by gregor** [comment]
Thanks for adding tests

**Pull request merged**